pkg/kubelet/status/status_manager.go

```diff
@@ -726,9 +726,14 @@ func GetSidecarsStatus(pod *v1.Pod) SidecarsStatus {
 				break
 			}
 		}
-		if !foundStatus && isSidecar {
-			klog.Infof("Pod %s: %s (sidecar): status not found, assuming not ready", format.Pod(pod), container.Name)
-			sidecarsStatus.SidecarsReady = false
+		if !foundStatus {
```

Comment:
Can we reverse ?
```
If foundStatus {
 break
}
```
And then do the two if.

Reply:
This would break us out of the loop on containers too early.